test/ductile/document_test.clj

```diff
@@ -662,6 +662,86 @@
                                              :refresh "true"})))
            "delete-by-query with wait-for-completion? set to false should directly return an answer before deletion with a task id")))))
 
+(deftest ^:integration update-by-query-test
+  (let [indexname "test_index"]
+    (for-each-es-version
+     "update by query."
+     #(es-index/delete! conn indexname)
+     (let [;; init state
+           doc-type (when (= 5 version) :test-type)
+           base-mappings (cond->> {:dynamic false ;; do not index fields without mapping
+                                   :properties {:name {:type "keyword"}
+                                                :age {:type "integer"}}}
+                           (= version 5) (assoc {} doc-type)) ;; ES5/7 mapping compatilbity
+           base-settings {:number_of_shards "1"
+                          :number_of_replicas "1"}
+           index-create-res (es-index/create!
+                             conn
+                             indexname
+                             {:mappings base-mappings
+                              :settings base-settings})
+           _ (assert (true? (boolean index-create-res))
+                     "the test index was not properly initialized")
+
+           ;; insert some documents
+           sample-docs (map #(hash-map :id (str (UUID/randomUUID))
+                                       ;; :_index indexname
+                                       ;; :_type doc-type
+                                       :name (str "name " %)
+                                       :age %
+                                       :sport "boxing")
+                            (range 20))
+           _ (doseq [doc sample-docs]
+               (sut/create-doc
+                conn
+                indexname
+                "doc-type"
+                doc
+                {:refresh "true"}))]
+       (testing "filter on a query and update with a script"
+         (sut/update-by-query
+          conn
+          [indexname]
+          {:script {:source "ctx._source.title='young'"}
+           :query {:range {:age {:lt 10}}}}
+          {:refresh "true"})
+
+         (is (= 10 (->> (sut/search-docs
+                         conn
+                         indexname
+                         {:query_string {:query "*"}}
+                         {} {})
+                        :data
+                        (filter #(-> % :title (= "young")))
+                        count))
+             "selected records have gotten updated"))
+       (testing "pick new properties"
+         (let [query-fn #(sut/query
+                          conn
+                          indexname
+                          {:match {"sport" "boxing"}} {})]
+           ;; for some reason ver 5 still retuns the results for the query above,
+           ;; even though it supposedly shouldn't. The mapping for the field doesn't
+           ;; exist at this point and it's not indexed
```

Comment:
I tested a similar scenario with both ES5 and ES7 the reason is not in Elasticsearch.

Reply:
I appreciate you going that far and testing each case manually. Indeed the "mysterious" problem that made me stuck turned out to be not a problem at all.